addons/spreadsheet/static/src/helpers/geo_json_service.js

```diff
@@ -0,0 +1,144 @@
+                .then((usStates) => {
+                    const mapping = {};
+                    for (const state of usStates) {
+                        mapping[normalizeFeatureName(state.name)] = state.code;
+                        mapping[normalizeFeatureName(state.code)] = state.code;
+                    }
+                    usaStatesMapping = mapping;
+                    usStatesMappingPromise = undefined;
+                    return mapping;
+                })
+                .catch((e) => {
+                    console.error(e);
+                    usaStatesMapping = {};
+                    usStatesMappingPromise = undefined;
+                    return {};
+                });
+            return usStatesMappingPromise;
+        }
+
+        async function getCountriesMapping() {
+            if (countriesMapping) {
+                return countriesMapping;
+            }
+            if (countriesMappingPromise) {
+                return countriesMappingPromise;
+            }
+            countriesMappingPromise = orm
+                .searchRead("res.country", [], ["name", "code"])
+                .then((resCountries) => {
+                    const mapping = {};
+                    for (const country of resCountries) {
+                        mapping[normalizeFeatureName(country.name)] = country.code;
+                        mapping[normalizeFeatureName(country.code)] = country.code;
+                    }
+                    countriesMapping = mapping;
+                    countriesMappingPromise = undefined;
+                    return mapping;
+                })
+                .catch((e) => {
+                    console.error(e);
+                    countriesMapping = {};
+                    countriesMappingPromise = undefined;
+                    return {};
+                });
+
+            return countriesMappingPromise;
+        }
+
+        async function fetchJsonFromServer(url) {
+            if (geoJsonCache.has(url)) {
+                return geoJsonCache.get(url);
+            }
+            if (geoJsonPromises.has(url)) {
+                return geoJsonPromises.get(url);
+            }
+
+            const promise = fetch(url, { method: "GET" })
+                .then((res) => res.json())
+                .then((geoJson) => {
+                    geoJsonCache.set(url, geoJson);
+                    geoJsonPromises.delete(url);
+                    return geoJson;
+                })
+                .catch((e) => {
+                    console.error(e);
+                    geoJsonCache.set(url, { type: "FeatureCollection", features: [] });
+                    geoJsonPromises.delete(url);
+                    return geoJsonCache.get(url);
+                });
+
+            geoJsonPromises.set(url, promise);
+            return promise;
+        }
+
+        const stateNameRegex = /(.*?)(\(.*\))?$/;
+
+        return {
+            getAvailableRegions: () => [
```

Comment:
our friends from south america are gonna be sad

Reply:
You're right, let's make this PR 50K lines 😎